packages/domain-picker/src/utils/domain-suggestions.ts

```diff
@@ -0,0 +1,31 @@
+type DomainSuggestion = import('@automattic/data-stores').DomainSuggestions.DomainSuggestion;
+
+// FIX ME: This code is duplicated within the domain-picker package
+export function getFreeDomainSuggestions( allSuggestions: DomainSuggestion[] | undefined ) {
+	return allSuggestions?.filter( ( suggestion ) => suggestion.is_free );
+}
+
+export function getPaidDomainSuggestions( allSuggestions: DomainSuggestion[] | undefined ) {
+	return allSuggestions?.filter( ( suggestion ) => ! suggestion.is_free );
+}
+
+// Recommend either an exact match or the highest relevance score
+export function getRecommendedDomainSuggestion( allSuggestions: DomainSuggestion[] | undefined ) {
+	if ( ! ( Array.isArray( allSuggestions ) && allSuggestions.length > 0 ) ) {
+		return;
+	}
+	const recommendedSuggestion = allSuggestions?.reduce( ( result, suggestion ) => {
+		if ( result.match_reasons && result?.match_reasons?.indexOf( 'exact-match' ) > -1 ) {
+			return result;
+		}
+		if ( suggestion.match_reasons && suggestion?.match_reasons?.indexOf( 'exact-match' ) > -1 ) {
+			return suggestion;
+		}
+		if ( suggestion.relevance > result.relevance ) {
+			return suggestion;
+		}
+		return result;
+	} );
+
+	return recommendedSuggestion;
+}
```

Comment:
I wonder if all of these could be selectors on the domain store 🤔

Reply:
It makes sense for this to be in the stores. I can only think that's there's just going to be more props going in to domain picker.

Reply:
I think the source of a large proportion of the props flowing to `domain-picker` is the mixed logic happening in `client/landing/gutenboarding/hooks/use-domain-suggestions.ts`. I feel the fetching call in there should belong to `domain-picker` package and any processing done to `allSuggestions` should be handled directly from here. In this way, we can pass just an optional prop `prioritisedSearch: string` if we want to overwrite the search term.

Reply:
I'm not opposed to this duplication for now. As we introduce shared usage, it may become clear where things should sit.

@razvanpapadopol those are good thoughts. What do you think of addressing them after the bulk move is landed in this PR?